### ISSUE TITLE
Cart state does not persist after using useApplyCartLinesChange or navigating checkout steps

### ISSUE THREAD

**Issue by pixelchutes** [Shopify/ui-extensions#448]
## Please list the package(s) involved in the issue, and include the version you are using

`@shopify/checkout-ui-extensions-react`: `v0.18.1`

## Describe the bug

We recently started to experience some regression with `useApplyCartLinesChange()` around `2022-08-12`, specifically [adding a new cart line](https://shopify.dev/api/checkout-extensions/checkout/extension-points/api#cartlineaddchange) results in removal of the very last cart line, _but only on subsequent attempts!_

```jsx

const applyCartLinesChange = useApplyCartLinesChange();

// ...

const result = await applyCartLinesChange({
  type: 'addCartLine',
  merchandiseId: 'gid://shopify/CartLine/123',
  quantity: 1,
});
```

☝️ I'm noticing that the new cart line is being added as expected, but seems to be replacing the very last cart line as a result, which has not been my experience prior. What's more is it does not seem to happen on the initial page load (Checkout), only after subsequent (2+) `applyCartLinesChange()` calls.

## Steps to reproduce the behavior:

1. Add a single (1) product to your cart, then proceed to Checkout
1. Ensure a product offer Checkout UI extension exists on each step of Checkout (`Checkout::Dynamic::Render`)
1. Start on the `information` step of Checkout _(specific step is irrelevant to reproduce, but useful to illustrate potentially related issue below)_
1. Add a new cart line by calling `applyCartLinesChange()`, passing a `CartLineChange` param with `type: 'addCartLine'` etc
1. Notice the new cart line is correctly added to the cart _(e.g. you should now see 2 cart lines in the Checkout UI)_
1. Now, add an additional cart line by calling `applyCartLinesChange()` again, this time for an alternate Product.
1. Notice the new cart line is added to the cart, but has now _replaced_ the previously added cart line from the previous attempt _(e.g. you should now see 3 cart lines in the Checkout UI, but only 2 remain.)_
1. Further, if you now proceed to the `shipping` step, the cart state does not persist -- it resets to the initial, single (1) cart line, or whatever the cart state was _on initial page load._

## Expected behavior

Subsequent calls to `applyCartLinesChange({type: 'addCartLine', ...})` should allow for multiple cart lines to be added without clobbering previously added lines.

Navigating through Checkout steps _(e.g. `information` > `shipping`)_ should persist cart lines and modified quantities, etc.

## Screenshots

https://www.loom.com/share/822c2ad0c23a430eb6c3c3d526fcfe09

**Comment by jeanffc** [comment]
Same here

**Comment by jzier3** [comment]
same

**Comment by eyal0s** [comment]
Same here

**Comment by thanigai-babu17** [comment]
facing the same issue in my case it adds the product to the cart items but if i click on "add" again instead of updating the quantity it does nothing and if try to add another product it just replaces the previously added product instead of appending it to the line items

**Comment by jaapvergote** [comment]
same for us

**Comment by meecrobe** [comment]
same here

**Comment by jarthorn** [comment]
We have released a fix for this issue! Can partners tracking this issue please comment to confirm it is working for your app, or especially if you are still seeing a problem here?

**Comment by jarthorn** [comment]
Great!! I'm going to mark this as fixed, but everyone is welcome to open a new issue if they still see any problems.
